ksql-serde/src/test/java/io/confluent/ksql/serde/json/KsqlJsonDeserializerTest.java

```diff
@@ -143,4 +140,21 @@ public void shouldTreatNullAsNull() throws JsonProcessingException {
     assertThat(genericRow, equalTo(expected));
 
   }
+
+  @Test
+  public void shouldCreateJsonStringForStructIfDefinedAsVarchar() throws JsonProcessingException {
+    final Schema schema = SchemaBuilder.struct()
+        .field("itemid".toUpperCase(), Schema.OPTIONAL_STRING_SCHEMA)
+        .build();
+    final KsqlJsonDeserializer deserializer = new KsqlJsonDeserializer(schema, false);
+    final Map<String, Object> row = new HashMap<>();
+    row.put("itemid", "{\"CATEGORY\":{\"ID\":2,\"NAME\":\"Food\"},\"ITEMID\":6,\"NAME\":\"Item_6\"}");
+
+    System.out.println(objectMapper.writeValueAsString(row));
```

Comment:
We probably shouldn't print to stdout here.

Reply:
Good catch! Forgot remove it.